url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/TeaLibrary.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Scripty Programming Language
+ * Copyright (C) 2010-2012 Bruno Ranschaert, S.D.I.-Consulting BVBA
+ * http://www.sdi-consulting.be
+ * mailto://info@sdi-consulting.be
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.cmdlib;
+
+import com.sdicons.scripty.parser.CommandException;
+import com.sdicons.scripty.annot.*;
+
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+
+@ScriptyStdArgList(name="text and password", fixed={@ScriptyArg(name="text", type="String")}, named = {@ScriptyArg(name="password", type="String", optional=true, value="")})
+public class TeaLibrary
+{
+    @ScriptyCommand(name="tea-encrypt")
+    @ScriptyRefArgList(ref = "text and password")
+    public static String encrypt(@ScriptyParam("text") String aPlainText, @ScriptyParam("password") String aPassword)
+    throws CommandException
+    {
+        TeaEncrypter lTea;
+
+        if(aPassword == null || aPassword.length() <= 0) lTea = new TeaEncrypter(); // Standard IBM EGL password.
+        else lTea = new TeaEncrypter(md5(aPassword)); // User provided password.
+
+        return lTea.encrypt(aPlainText);
+    }
+
+    @ScriptyCommand(name="tea-decrypt")
+    @ScriptyRefArgList(ref = "text and password")
+    public static String decrypt(@ScriptyParam("text") String aCipher, @ScriptyParam("password") String aPassword)
+    throws CommandException
+    {
+        TeaEncrypter lTea;
+
+        if(aPassword == null  || aPassword.length() <= 0) lTea = new TeaEncrypter(); // Standard IBM EGL password.
+        else lTea = new TeaEncrypter(md5(aPassword)); // User provided password.
+
+        return lTea.decrypt(aCipher);
+    }
+
+    private static byte[] md5(String aString)
+    throws CommandException
+    {
+        try
+        {
+            MessageDigest digest = java.security.MessageDigest.getInstance("MD5");
+            digest.update(aString.getBytes());
+            return digest.digest();
+        }
+        catch (NoSuchAlgorithmException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)